includes/modules/pages/search_result/header_php.php

```diff
@@ -2,11 +2,14 @@
 /**
  * Header code file for the Search Results page
  *
- * @copyright Copyright 2003-2023 Zen Cart Development Team
+ * @copyright Copyright 2003-2022 Zen Cart Development Team
  * @copyright Portions Copyright 2003 osCommerce
  * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
- * @version $Id: pRose on charmes 2023 Feb 03 Modified in v1.5.8a $
+ * @version $Id: DrByte 2020 Dec 25 New in v1.5.8-alpha $
```

Comment:
it looks like you are making a PR against the v1.5.8-alpha tag.  i believe all PRs should be going against the master branch.

IMO, changes in the stamp should only be done via the repo maintainers, and then when they are doing some sort of new tag or release.  otherwise it can get very confusing when doing upgrades.  if a PR changes the stamp, i would surmise something is a bit adrift with your process.

Reply:
I thought this work was development on v158, sorry.  I have rebased my change onto upstream/master and force pushed (I believe a force push is required in this case, a simple push was rejected by git).

Reply:
no apologies necessary.

unfortunately, this still looks wrong.  i would refer to #5911 with regards to how PRs should work.

with regards to the `$Id` and other elements in the stamp, how they work is not important (at least to me).  i have opened issues and there have been discussions on them in the past, and i am comfortable with whatever process the maintainers use.  the point that i am making is that if your PR changes the stamp (and yours does), then there is a problem with your process of creating that PR.  again, i would refer to the above mentioned post.

however, in doing a cursory look, it does seem that your new class file, when using existing notifiers, those notifiers have the correct signature, ie they are the same as in the master branch; and not what is in the v1.5.8a tag.

as an example:

```
charmes: # git checkout v1.5.8a-alpha 
Note: switching to 'v1.5.8a-alpha'.

You are in 'detached HEAD' state
....
charmes: # git diff master includes/modules/pages/search_result/header_php.php
...
// Notifier Point
-$zco_notifier->notify('NOTIFY_SEARCH_COLUMNLIST_STRING', $select_column_list, $select_column_list);
+$zco_notifier->notify('NOTIFY_SEARCH_COLUMNLIST_STRING');

...
 
 // Notifier Point
-$zco_notifier->notify('NOTIFY_SEARCH_SELECT_STRING', $select_str, $select_str);
+$zco_notifier->notify('NOTIFY_SEARCH_SELECT_STRING');

```
in the above section of code, the added parameter elements to the notifiers are in the master branch, while the notifiers without parameters on in the v1.5.8a-alpha tag.  and when i look at your new class file for those notifiers, the parameters are there:

https://github.com/zencart/zencart/blob/86e2bd9690e46aaa9e61e26b88992ec03f592024/includes/classes/class.search.php#L198

https://github.com/zencart/zencart/blob/86e2bd9690e46aaa9e61e26b88992ec03f592024/includes/classes/class.search.php#L210

i'm not sure what more to say on it...

hopefully it makes sense to you on it.  and again, i am a fan of what you are doing here.

Reply:
I'm a little confused as to how those lines are wrong.  Github 'show files' on my PR shows, for example, the COLUMNLIST_STRING one looks like (from https://github.com/zencart/zencart/pull/6036/files) :

`        $this->notify('NOTIFY_SEARCH_COLUMNLIST_STRING', $select_column_list, $select_column_list);`

This is correct, no?  If you go back to the v1.5.8.a-alpha tag then the two parameters are missing, yes, but that's old code and was improved apparently by yourself at https://github.com/zencart/zencart/commit/fd0820894fbc1173e705bb515e625a9f29653524

Visible in latest code on `master`: https://github.com/zencart/zencart/blob/9ae0b145fff3d89bbeafdae83b1f9c00dbdf6788/includes/modules/pages/search_result/header_php.php#L211

Can you spell out what's wrong in a little more detail please?

Reply:
yes, the code looks correct.  (i have not tested it.)

you based your changes on the wrong branch; which is why the header comments have changed.  somehow you seem to have done the code correctly.  the commit you referenced, fd08208, was done 11/23/2022.  and yet the header that you are now changing is back to a release on 7/13/2022.

if you are confident that the code is correct, and no one else has a problem, i would not worry about it.  

if you still do no understand what i am saying, perhaps someone else can help clarify it.

Reply:
Well, "check twice, merge once" as someone somewhere probably says... Upon diffing against the original `search_results/php_header.php` I've found a couple of places `$_GET` was still used in the refactored code.  I've fixed these up to use the `$searchOptions` object as they should, but not committed or pushed yet, because of the next point.

I also note, the new class.search.php file has no Copyright comment at the top or place for the `$Id blah$` string. @drbyte could you please say what I should put in a brand new file?  Is it enough to have something like:

```
/**
 * @copyright Copyright 2003-2023 Zen Cart Development Team
 * @license http://www.zen-cart.com/license/2_0.txt GNU Public License V2.0
 * @version $Id:  $
 */
```

and something will fill in the `$Id$` bit automatically?  I've always seen people's username, a date and some kind of tag id in there but never known where they come from or how they get updated; I presume it's not manually typed in by someone.

Reply:
Due to lack of response, I've copied the copyright header from `search_result/header_php.php` (where I refactored most of the code from to make `class.search.php`) and force pushed my branch, making sure it's rebased onto upstream/master.  I presume the `Id` tags will sort themselves out in the merge.

Reply:
Yes, the basic copyright header which you copied over is fine. The `@version $Id: $` string gets updated when we run the script to stamp all files shortly before tagging a release. Until that stamping it's just the same as the previous stamping.